libs/pika/async_mpi/include/pika/async_mpi/mpi_polling.hpp

```diff
@@ -182,53 +203,43 @@ namespace pika::mpi::experimental {
+    PIKA_EXPORT void set_enable_pool(bool);
 
     // -----------------------------------------------------------------
-    /// Get the polling transfer mode for continuations.
+    /// Set/Get the polling and handler mode for continuations.
     PIKA_EXPORT std::size_t get_completion_mode();
-    PIKA_EXPORT void set_completion_mode(std::size_t mode);
-
-    PIKA_EXPORT bool create_pool(
-        std::string const& = "", pool_create_mode = pool_create_mode::pika_decides);
-
-    PIKA_EXPORT const std::string& get_pool_name();
-    PIKA_EXPORT void set_pool_name(const std::string&);
-
-    // returns false if no custom mpi pool has been created
-    PIKA_EXPORT bool pool_exists();
-
-    PIKA_EXPORT void register_polling();
+    PIKA_EXPORT void set_completion_mode(std::size_t);
 
+    /// can be used to choose between single/multi thread model when initializing MPI
     PIKA_EXPORT int get_preferred_thread_mode();
 
-    /// when true pika::mpi can disable the pool, or change the thread mode
+    /// when true pika::mpi can disable pool creation, or change the thread mode
     /// otherwise, the flags passed by the user to completion mode etc are honoured
-    PIKA_EXPORT void enable_optimizations(bool enable);
+    PIKA_EXPORT void enable_optimizations(bool);
+
+    /// return the name assigned to the mpi polling pool
+    PIKA_EXPORT const std::string& get_pool_name();
 
     // initialize the pika::mpi background request handler
     // All ranks should call this function (but only one thread per rank needs to do so)
-    PIKA_EXPORT void init(bool init_mpi = false, bool init_errorhandler = false);
+    PIKA_EXPORT void start_polling(
+        exception_mode errorhandler = no_handler, std::string pool_name = "");
 
-    // -----------------------------------------------------------------
-    PIKA_EXPORT void finalize(std::string const& pool_name = "");
+    PIKA_EXPORT void stop_polling();
 
     // -----------------------------------------------------------------
-    // This RAII helper class assumes that MPI initialization/finalization is handled elsewhere
-    struct [[nodiscard]] enable_user_polling
+    // This RAII helper class ensures that MPI polling start/stop is handled correctly
+    struct [[nodiscard]] enable_polling
     {
-        enable_user_polling(std::string const& pool_name = "", bool init_errorhandler = false)
-          : pool_name_(pool_name)
+        /// an empty pool name tells pika to use the mpi pool if it exists
+        explicit enable_polling(
+            exception_mode errorhandler = no_handler, std::string const& pool_name = "")
         {
-            mpi::experimental::init(false, init_errorhandler);
-            mpi::experimental::register_polling();
+            start_polling(errorhandler, pool_name);
         }
 
-        ~enable_user_polling() { mpi::experimental::finalize(pool_name_); }
-
-    private:
-        std::string pool_name_;
+        ~enable_polling() { stop_polling(); }
```

Comment:
I think most/all of this can be in `detail`?

`start/stop/enable_polling` could arguably stay in non-detail I suppose. However, I still think we could enable polling by default _or_ start polling the first time `transform_mpi` is used. What do you think?

Reply:
I've moved everything that isn't used by DLAF or by a unit test (for an explicit feature) into detail